dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp

```diff
@@ -561,7 +561,8 @@ LearnerReadSnapshot DAGStorageInterpreter::doBatchCopLearnerRead()
         {
             // We can also use current thread to resolve lock, but it will block next process.
             // So, force this region retry in another thread in CoprocessorBlockInputStream.
-            force_retry.emplace(e.region_id);
+            for (const auto & lock : e.locks)
+                force_retry.emplace(lock.first);
```

Comment:
This will only set the region id into `region_retry_from_local_region` and swallowed up the `LockInfo`, making `FlashService::EstablishDisaggTask` only get `RegionException` but not `LockException`.
Maybe we need another function for disagg read.